tasks/service/impl/src/main/java/org/projectnessie/nessie/tasks/service/impl/TasksServiceImpl.java

```diff
@@ -114,11 +115,16 @@ <T extends TaskObj, B extends TaskObj.Builder> CompletionStage<T> submit(
     Obj obj = persist.getImmediate(taskRequest.objId());
     if (obj != null) {
       T taskObj = castObj(taskRequest, obj);
-      TaskStatus status = taskObj.taskState().status();
+      TaskState taskState = taskObj.taskState();
+      TaskStatus status = taskState.status();
       switch (status) {
         case FAILURE:
-          metrics.taskHasFinalFailure();
-          return failedStage(taskRequest.behavior().stateAsException(taskObj));
+          if (canRetryNow(taskState)) {
```

Comment:
This change deserves a change in `org.projectnessie.nessie.tasks.api.TaskObjUtil#TASK_DEFAULT_CACHE_EXPIRE` as well. And likely a test case w/ the persist-cache enabled.

Reply:
removed this from current PR... will follow-up separately.